src/v/archival/archiver_service.cc

```diff
@@ -0,0 +1,999 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "archival/archiver_service.h"
+
+#include "archival/logger.h"
+#include "archival/ntp_archiver_service.h"
+#include "archival/upload_housekeeping_service.h"
+#include "cloud_storage/cache_service.h"
+#include "cluster/partition_manager.h"
+#include "cluster/types.h"
+#include "config/configuration.h"
+#include "container/fragmented_vector.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "ssx/auto_fmt.h"
+#include "ssx/semaphore.h"
+#include "ssx/sformat.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/loop.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/util/later.hh>
+
+#include <boost/msm/back/state_machine.hpp>
+#include <boost/msm/front/euml/common.hpp>
+#include <boost/msm/front/functor_row.hpp>
+#include <boost/msm/front/state_machine_def.hpp>
+#include <boost/test/tools/old/interface.hpp>
+#include <fmt/ostream.h>
+
+#include <exception>
+#include <optional>
+#include <utility>
+
+namespace archival {
+
+// This file contains implementation of the managed_partition_stm.
```

Comment:
I suggest we don't overload the STM concept inside Redpanda with one other meaning. Maybe `managed_partition_fsm` instead? FSM=Finite State Machine.

We could also rename all existing uses of STM to RSM (replicated state machine) but that term is already too omnipresent in code, docs, chats, etc